crates/tabby-index/src/code/index.rs

```diff
@@ -165,6 +172,48 @@ fn is_valid_file(file: &SourceCode) -> bool {
         && file.number_fraction <= MAX_NUMBER_FRACTION
 }
 
+fn create_document_with_commit(doc: &TantivyDocument, commit: &str) -> TantivyDocument {
+    let schema = tabby_common::index::IndexSchema::instance();
+    doc! {
+        schema.field_id => get_text(doc, schema.field_id),
+        schema.field_source_id => get_text(doc, schema.field_source_id).to_string(),
+        schema.field_corpus => get_text(doc, schema.field_corpus).to_string(),
+        schema.field_attributes => json!({
+            code::fields::ATTRIBUTE_COMMIT: commit,
+        }),
+        schema.field_updated_at => get_text(doc, schema.field_updated_at).to_string(),,
+        schema.field_failed_chunks_count => get_json_number_field(doc, schema.field_failed_chunks_count, code::fields::FAILED_CHUNKS_COUNT) as usize,
```

Comment:
https://github.com/TabbyML/tabby/blob/c8735f5690ef82fd36afd38597eb966c207861df/crates/tabby-common/src/index/mod.rs#L64

field_failed_chunks_count is not a json field.